pkg/tfvars/vsphere/vsphere.go

```diff
@@ -55,6 +60,19 @@ func TFVars(sources TFVarsSources) ([]byte, error) {
 	// /<datacenter>/vm/<folder_path> so we can split on "vm/".
 	folderRelPath := strings.SplitAfterN(controlPlaneConfig.Workspace.Folder, "vm/", 2)[1]
 
+	// Must use the Managed Object ID for a port group (e.g. dvportgroup-5258)
+	// instead of the name since port group names aren't always unique in vSphere.
+	// https://bugzilla.redhat.com/show_bug.cgi?id=1918005
+	networkID, err := getNetworkMoID(controlPlaneConfig.Workspace.Server,
+		sources.Username,
+		sources.Password,
+		controlPlaneConfig.Workspace.Datacenter,
+		sources.Cluster,
+		controlPlaneConfig.Network.Devices[0].NetworkName)
```

Comment:
Are there any corresponding changes to the machine-api-operator's vSphere provider to ensure that workers use the correct network as well?

Reply:
Yes, SPLAT-345 is where we will track those changes. I'll cross link in this PR too.